ddtrace/llmobs/_integrations/utils.py

```diff
@@ -149,14 +149,18 @@ def extract_message_from_part_google(part, role=None):
         message["role"] = role
     if function_call:
         function_call_dict = function_call
-        if not isinstance(function_call, dict):
+        if hasattr(function_call, "model_dump"):
```

Comment:
```suggestion
        if hasattr(function_call, "model_dump") and callable(obj.model_dump):
```
Just to be sure model_dump is a function we can call